app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -30,8 +30,17 @@ def verify_credentials(_auth_type = nil, options = {})
         # As the connect will only give a handle
         # we verify the credentials via an actual operation
         connect(options).inventory.list_feeds
+      rescue URI::InvalidComponentError
+        raise MiqException::MiqHostError, "Host '#{hostname}' is invalid"
```

Comment:
under which circumstances this error (Host X is invalid) will occur? In other words, what does 'invalid' mean?